osu.Game/Screens/Select/DifficultyRecommender.cs

```diff
@@ -72,6 +85,26 @@ private void calculateRecommendedDifficulties()
             });
         }
 
+        private IEnumerable<RulesetInfo> bestRulesetOrder;
+
+        private IEnumerable<RulesetInfo> getBestRulesetOrder()
+        {
+            if (bestRulesetOrder != null)
+                return bestRulesetOrder;
+
+            var otherRulesets = recommendedStarDifficulty.ToList()
+                                                         .Where(pair => !pair.Key.Equals(ruleset.Value))
```

Comment:
To achieve the effect I assume is desired here, you could actually go a little weird to make this shorter and leverage `OrderBy`:
```csharp
> new[] { false, false, true, false }.OrderByDescending(x => x)
OrderedEnumerable<bool, bool> { true, false, false, false }
```
and then chain with `ThenByDescending` to get the secondary ordering.